src/components/AppConfig/HealthCheck.tsx

```diff
@@ -0,0 +1,119 @@
+import React from 'react';
+
+import { HealthCheckResult } from "@grafana/runtime";
+import { Alert, AlertVariant } from "@grafana/ui";
+
+interface HealthCheckDetails {
+  openAI: OpenAIHealthDetails | boolean;
+  vector: VectorHealthDetails | boolean;
+  version: string;
+}
+
+interface OpenAIHealthDetails {
+  configured: boolean;
+  ok: boolean;
+  error?: string;
+  models: Record<string, OpenAIModelHealthDetails>;
+}
+
+interface OpenAIModelHealthDetails {
+  ok: boolean;
+  error?: string;
+}
+
+interface VectorHealthDetails {
+  enabled: boolean;
+  ok: boolean;
+  error?: string;
+}
+
+const isHealthCheckDetails = (obj: unknown): obj is HealthCheckDetails => {
+  return typeof obj === 'object' && obj !== null && 'openAI' in obj && 'vector' in obj && 'version' in obj;
+}
+
+const alertVariants = new Set<AlertVariant>(['success', 'info', 'warning', 'error']);
+const isAlertVariant = (str: string): str is AlertVariant => alertVariants.has(str as AlertVariant);
+const getAlertVariant = (status: string): AlertVariant => {
+  if (status.toLowerCase() === 'ok') {
+    return 'success';
+  }
+  return isAlertVariant(status) ? status : 'info';
+};
+const getAlertSeverity = (status: string, details: HealthCheckDetails): AlertVariant => {
+  const severity = getAlertVariant(status);
+  if (severity !== 'success') {
+    return severity;
+  }
+  if (!isHealthCheckDetails(details)) {
+    return 'success';
+  }
+  if (typeof details.openAI === 'object' && typeof details.vector === 'object') {
+    const vectorOk = !details.vector.enabled || details.vector.ok;
+    return details.openAI.ok && vectorOk ? 'success' : 'warning';
+  }
+  return severity;
+}
+
+export function ShowHealthCheckResult(props: HealthCheckResult) {
+  let severity = getAlertVariant(props.status ?? 'error');
+  if (!isHealthCheckDetails(props.details)) {
+    return (
+      <div className="gf-form-group p-t-2">
+        <Alert severity={severity} title={props.message}>
+        </Alert>
+      </div>
+    );
+  }
+
+  severity = getAlertSeverity(props.status ?? 'error', props.details);
+  const message = severity === 'success' ? 'Health check succeeded!' : props.message;
+
+  return (
+    <div className="gf-form-group p-t-2">
+      <Alert severity={severity} title={message}>
+        <ShowOpenAIHealth openAI={props.details.openAI} />
+        <ShowVectorHealth vector={props.details.vector} />
+      </Alert>
```

Comment:
Thinking about the scenario where the OpenAI is correctly configured but vectorDB is ignored or misconfigured, probably it would be clearer to have two alerts, one per technology. WDYT?

Reply:
Yep, makes sense and looks nicer, thanks!